src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,147 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_LOAD_FACTOR * table.length);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = getHash(key);
```

Comment:
Let's look at another implementation
```suggestion
        public void put(K key, V value) {
        if (size >= threshold) {
            resizeTables();
        }
        Node<K, V> newNode = new Node<>(key, value);
        int bucketIndex = getBucketIndex(key);
        if (table[bucketIndex] == null) {
            table[bucketIndex] = newNode;
            size++;
        } else {
            linkNode(table[bucketIndex], newNode);
        }
    }
```

Reply:
I saw these implementations, thanks. and I changed some methods